Chap_API_Storage.tex

```diff
@@ -0,0 +1,90 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Chapter: Storage support
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\chapter{Storage Support Definitions}
+\label{chap:api_storage}
+
+Distributed and parallel computing systems are increasingly embracing storage hierarchies to meet the diverse data management needs of applications and other systems software in a cost-effective manner.
+These hierarchies provide access to a number of distinct storage layers, with each potentially composed of different storage hardware (e.g., HDD, SDD, tape, NVM), deployed at different locations (e.g., on-node, on-switch, on-site, WAN), and designed using different storage paradigms (e.g., file-based, object-based).
+Each of these systems offers unique performance and usage characteristics that storage system users should carefully consider to ensure the most efficient use of storage resources.
+
+PMIx enables users to better understand storage hierarchies by defining attributes that formalize storage system characteristics, state, and other parameters.
+These attributes can be queried by applications, I/O libraries and middleware, and workflow systems to discover available storage resources and to inform on which resources are most suitable for different I/O workload requirements.
+
+
+%%%%%%%%%%%
+\section{Storage-related attributes}
+\label{api:struct:attributes:pstrg}
+
+\versionMarker{4.0}
+
+The following attributes may be returned in response to queries (e.g., \refapi{PMIx_Get} or \refapi{PMIx_Query_info}) made by processes or tools.
+
+%
+\declareAttributeNEW{PMIX_STORAGE_ID}{"pmix.strg.id"}{char*}{
+An identifier for the storage system (e.g., lustre-fs1, daos-oss1, home-fs)
+}
+
+%
+\declareAttributeNEW{PMIX_STORAGE_PATH}{"pmix.strg.path"}{char*}{
```

Comment:
I believe this is only used as a qualifier, yes? I would so indicate, if that is true.

Reply:
Same as above, though I think this one's more immediately useful -- What is the mount path corresponding to a given PMIX_STORAGE_ID? You could imagine querying PMIX_QUERY_STORAGE_LIST to get a list of storage systems PMIx is aware of, then getting their mount points so you could do I/O to them, for instance.